test/Market.t.sol

```diff
@@ -265,37 +264,22 @@ contract Market_replace_reverts_ReplacementAmountExceedsNrtDeficit is
   function setUp() external {
     _listRemovals();
     _createCertificate();
-
     _market.grantRole({
       role: _market.MARKET_ADMIN_ROLE(),
       account: _namedAccounts.admin
     });
-    _removal.grantRole({
-      role: _removal.RELEASER_ROLE(),
-      account: _namedAccounts.admin
-    });
-    uint256 amount = _market.calculateCheckoutTotal(_amountToReplace);
-    vm.startPrank(_namedAccounts.admin);
-    assertEq(_certificate.getNrtDeficit(), 0); // no discrepancy
-    _bpNori.deposit(_namedAccounts.admin, abi.encode(amount));
-    _bpNori.approve(address(_market), amount);
   }
 
   function test() external {
-    // using low level call to get around limitation of vm.expectRevert
```

Comment:
@amiecorso what was the original context behind why this was needed here?

Reply:
interesting.... so the limitation of `vm.expectRevert` is that it applies to the _very next function call_ -- so if you're wanting to know whether `replace` is going to revert, that becomes a problem if there are other function calls that happen in the execution of `replace` before the revert you expect (and if those *don't* revert, then `vm.expectRevert` will fail).  the approach here uses the low-level call to get a success/failure code, and then checks that directly -- so it's just a hackier way to ask the question you're trying to ask without relying on `vm.expectRevert`

I'm not sure what changed to allow it to work with `expectRevert` now -- maybe they fixed the limitation of `vm.expectRevert`? (that'd be cool) maybe something in our code changed?

Reply:
maybe here?: https://github.com/foundry-rs/foundry/pull/4945

Reply:
problem with the hacky workaround was that we weren't actually testing what caused the revert which made the test fragile and the title misleading. Useful hack, but we should be careful if we need to rely on anything like it in the future and make sure we aren't getting a revert for other reasons.